product/v2.py

```diff
@@ -240,6 +261,30 @@ def upload_component_descriptor_v2_to_oci_registry(
     else:
         raise NotImplementedError(on_exist)
 
+    blobs = []
+    if source_ref:
+        manifest = client.manifest(
+            image_reference=source_ref
+        )
+
+        # download and reupload all layers except component descriptor layer
+        for layer in manifest.layers:
+            if not layer.mediaType == gci.oci.component_descriptor_mimetype:
```

Comment:
stylistic nit: I would prefer to invert the check and use a `continue`, thus reducing the indentation of the actual processing